### PR TITLE
ToolButtonMenu for switching between Note Entry Modes

### PR DESCRIPTION
**Pull request by shoogle**
Adds a dropdown menu for choosing note entry mode.

image

The ToolButtonMenu class is useful for grouping related actions under a single toolbar icon. It could be used to consolidate other actions in the future. For example:

- Augmentation dots
- Durations
- Voices

A ToolButtonMenu for augmentation dots would allow the user to choose how many dots to add, with the default being one.

### CODE REVIEW

**Comment by shoogle** [comment]
**Please don't merge unless this is fully rebased and fully up-to-date with the master branch!**
This is to ensure that the PR's commit hash is preserved in the master branch for authenticity purposes.

File: mscore/musescore.qrc [hunk 1]
```diff
@@ -27,7 +27,6 @@
         <file>data/icons/png/window-close-hover@2x.png</file>
         <file>data/icons/png/window-float.png</file>
         <file>data/icons/png/window-float@2x.png</file>
-        <file>revision.h</file>
```
Comment (Jojo-Schmitz) [inline mscore/musescore.qrc]: this file is needed by musescore.cpp, line 5184 (before this commit)

Comment (shoogle) [inline mscore/musescore.qrc]: Oops, good catch! I was testing something but forgot to put it back afterwards.

**Comment by lasconic** [comment]
I merged this. Be aware that it breaks repitch mode. So looking forward to your next PR to fix that and implement other modes.

**Comment by shoogle** [comment]
It looks like repitch mode was actually broken by 1a8cdf1da4ea63b851001feddf37bb4075c4446c. Might be a good idea to have a test for repitch mode. I'll make one when I get around to making tests for the other modes.

**Comment by mscscf** [comment]
I pulled MuseScore-master from GitHub.

I have MuseScore2 version 2.1.0 installed.

Do I need to install MuseScore-master to get the new dropdown?

**Comment by lasconic** [comment]
No, the dropdown is already in MuseScore 2.1

**Comment by mscscf** [comment]
Do you know what I need to do to get the new Note Input options to show?  The 3 new Note Input options do not show preferences.

**Comment by MarcSabatella** [comment]
If you have 2.1 they are there. Be sure you really have 2.1 - see Help /
About. And then look closely just to the right of the "N" icon on the note
input toolbar - that's where the drop down is. It's not in Preferences.

**Comment by mscscf** [comment]
image

Where is the "N" icon?

**Comment by mscscf** [comment]
image

**Comment by Jojo-Schmitz** [comment]
See https://musescore.org/en/handbook-basics/note-input#other-input-modes

**Comment by mscscf** [comment]
Ok thanks for your help.  I see them now.  I'll give them a try.  

Somehow I got the idea the options were in preferences.  Not just on the main screen.

**Comment by mscscf** [comment]
How can I pick up the nightly builds instead of waiting for version 3.0?  In case there are fixes in the nightly builds.

**Comment by MarcSabatella** [comment]
Nightly builds for 2.x are available on the Downloads page of the main
site. Don't use "master" builds, they are extremely unstable and
unsupported.

Also for future reference, it is best to ask for help on the Support forum.

**Comment by mscscf** [comment]
Ok thanks.  If I need more info on getting the nightly builds, I'll look in
the forum.  I saw comments that the rhythm option was broken on this page.
Hopefully that is fixed.

**Pull request merged**